### ISSUE TITLE
Cannot override max_complexity in Schema.execute anymore

### ISSUE THREAD

**Issue by riffraff** [rmosolgo/graphql-ruby#774]
I just upgraded from 1.4.x to 1.6 and it seems this code 

```
opts = {max_complexity: 300}
MySchema.execute(query, opts)
```

does not override max_complexity anymore as it previously did.

As far as I can tell, this is because we have this sequence:
```
def execute(query_str = nil, **kwargs) # kwargs is {max_complexity: 300}
...  # invokes
multiplex([kwargs])  # sends [{max_complexity: 300]
... # gets to
def multiplex(*args) # args is [[{max_complexity: 300}]]
... # invokes 
GraphQL::Execution::Multiplex.run_all(self, *args)
... # in there
def run_all(schema, query_options, *rest)
  queries = query_options.map { |opts| GraphQL::Query.new(schema, nil, opts) }
  run_queries(schema, queries, *rest)
end
... # query_options is [{max_complexity: 300}] but "rest" is [], Query gets created with these options, but run_queries expects a max_complexity which can't be set
def run_queries(schema, queries, context: {}, max_complexity: nil)
```

Maybe I just need to change the way I invoke `MySchema.execute` but I can't see a way to do this.

**Comment by rmosolgo** [comment]
I moved max-complexity from query-level to multiplex level but it looks like I made a mistake in doing so! I think _either_ 

- `max_complexity` should be extracted from `Schema.execute`'s `opts` and passed as a multiplex option; OR 
- `max_complexity` should _still_ be accepted as a query-level option, and the analyzer should apply the override to that query only.

**Comment by riffraff** [comment]
It seems to me the override should be per execute, but I don't really have a clear understanding of the whole multiplexing use case :)

**Comment by esjee** [comment]
Hello,

I tried looking into this, but I don't believe I'm seeing the same behaviour as described. I tried writing a test, expecting it to fail, but it seems to succeed: https://github.com/rmosolgo/graphql-ruby/pull/800

@riffraff @rmosolgo Am I missing something or has this been fixed in the meantime?

**Comment by rmosolgo** [comment]
🤔  I don't _know_ that it's been fixed! What else could it be ... 


Looking back at the description: 

```
... # query_options is [{max_complexity: 300}] but "rest" is [], Query gets created with these options, but run_queries expects a max_complexity which can't be set
def run_queries(schema, queries, context: {}, max_complexity: nil)
```

That `run_queries(... max_complexity: nil)` is meant to be used with `Schema#multiplex`. 

@riffraff can you describe what your failure looks like? For example, is there a query that you think should be rejected, but it's allowed?

**Comment by riffraff** [comment]
I'm running the introspection query from GraphiQL, the schema has a default max complexity  of 100, but if it's the introspection query I allow 300, and the error I get is

```
{
  "errors": [
    {
      "message": "Query has complexity of 179, which exceeds max complexity of 100"
    },
    {
      "message": "Query has complexity of 179, which exceeds max complexity of 100"
    }
  ]
}
```
(not sure why the message occurs twice).

I'll try to see if this happens with current graphql-ruby master.

**Comment by riffraff** [comment]
nope, still happens with master for me.

**Comment by riffraff** [comment]
indeed, the problem seems to be happening only on increasing, see
https://github.com/rmosolgo/graphql-ruby/pull/804/files

**Comment by rmosolgo** [comment]
A failing test! Thank you! I'll take a look soon!

**Comment by esjee** [comment]
Now that there's a failing spec, I'd like to try to fix this (again)!

**Comment by rmosolgo** [comment]
Fixed by #812
